### PR TITLE
ENCD-5579 Update Dataset Search button types

### PR DESCRIPTION
**Pull request by forresttanaka**
I moved the collection of the dataset types actually included in the cart to the parent component, because now two child components need that information.

### CODE REVIEW

File: src/encoded/static/components/cart/cart.js [hunk 1]
```diff
@@ -943,9 +941,16 @@ const CartTools = ({ elements, selectedTerms, selectedDatasetType, typeChangeHan
     // datasets."
     const disabledMessage = selectedDatasetType === DEFAULT_DATASET_TYPE ? 'Select dataset type to download' : '';
     const selectedType = datasetTypes[selectedDatasetType].type;
+
+    // Make a list of all the dataset types currently in the cart.
+    const usedDatasetTypes = [Object.keys(defaultDatasetType)[0]].concat(elements.reduce((types, elementAtId) => {
+        const type = atIdToType(elementAtId);
+        return types.includes(type) ? types : types.concat(type);
+    }, []));
```
Comment (emro) [inline src/encoded/static/components/cart/cart.js:949]: I don't quite follow the distinction between "usedDatasetTypes" and "actualDatasetTypes" - why can't we compute "actualDatasetTypes" here and use it for both Cart components? CartTypeSelector needs the "all" type?

**Review by emro — APPROVED** [review]
The code all looks good! I would move to QA Ready but you made a note on the ticket about trying to get the Indexer test to pass. Should I hold off then?

**Comment by forresttanaka** [comment]
Seems like we’re allowing the indexer test to fail for now.

File: src/encoded/static/components/cart/cart.js (hunk 1 above)
Comment (forresttanaka) [reply src/encoded/static/components/cart/cart.js:949]: Yeah, the only difference is that `usedDatasetTypes` includes "all" while `actualDatasetTypes` does not.

**Pull request merged**